### PR TITLE
Add error handling

### PR DESCRIPTION
**Pull request by mestery**
Add some initial error handling into the CI scripts

Closes Issue #69

Signed-off-by: Kyle Mestery mestery@mestery.com

### CODE REVIEW

**Comment by mestery** [comment]
This needs a bit more work, but it's a start. Seeing how the travis-ci job run looks now.

File: ci/docker-rally.sh [hunk 1]
```diff
@@ -4,6 +4,9 @@
 XTRACE=$(set +o | grep xtrace)
 set -o xtrace
 
+# Track failures
+FAILED=0
```
Comment (huikang) [inline ci/docker-rally.sh:8]: Is this variable used to indicate how many tests the CI job fails?

Comment (mestery) [inline ci/docker-rally.sh:8]: It's mainly used to detect if a specific command in docker-rally.sh fails. If run fails, we want to ensure we run cleanup, but we need to remember run failed so we can return that to travis-ci. Make sense?

File: ci/docker-rally.sh [hunk 2]
```diff
@@ -12,13 +15,15 @@ echo "OVS_REPO=${OVS_REPO} OVS_BRANCH=${OVS_BRANCH}"
 # A combined script to run all the things
 
 # Prepare the environment
-./prepare.sh
+./prepare.sh || FAILED=$(( $FAILED + 1 ))
 
 # Run the testsuite
-./scale-run.sh $OVS_REPO $OVS_BRANCH
+./scale-run.sh $OVS_REPO $OVS_BRANCH || FAILED=$(( $FAILED + 1 ))
```
Comment (huikang) [inline ci/docker-rally.sh:21]: With the PR, shall we avoid calling exit in the other script, e.g., https://github.com/openvswitch/ovn-scale-test/blob/master/ci/scale-run.sh#L23

Comment (mestery) [inline ci/docker-rally.sh:21]: Yes, that should be a return instead of an exit. I will fix it.

Comment (huikang) [inline ci/docker-rally.sh:21]: thanks

**Comment by huikang** [comment]
@mestery is the PR ready for review or more to add? Thanks.

**Comment by mestery** [comment]
@huikang I think so, I plan to add more error backout in the other scripts, but this can merge as-is now. Thanks!

**Pull request merged**